service/src/main/java/org/cbioportal/service/impl/AlterationEnrichmentServiceImpl.java

```diff
@@ -0,0 +1,71 @@
+package org.cbioportal.service.impl;
+
+import org.cbioportal.model.*;
+import org.cbioportal.model.util.QueryElement;
+import org.cbioportal.model.util.Select;
+import org.cbioportal.service.AlterationEnrichmentService;
+import org.cbioportal.service.AlterationCountService;
+import org.cbioportal.service.util.AlterationEnrichmentUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Service
+public class AlterationEnrichmentServiceImpl implements AlterationEnrichmentService {
+
+    @Autowired
+    private AlterationCountService alterationCountService;
+    @Autowired
+    private AlterationEnrichmentUtil<AlterationCountByGene> alterationEnrichmentUtil;
+
+    @Override
+    public List<AlterationEnrichment> getAlterationEnrichments(
+        Map<String, List<MolecularProfileCaseIdentifier>> molecularProfileCaseSets, final Select<MutationEventType> mutationEventTypes,
+        final Select<CopyNumberAlterationEventType> cnaEventTypes, EnrichmentType enrichmentType) {
+
+        Map<String, List<AlterationCountByGene>> alterationCountsbyEntrezGeneIdAndGroup = getAlterationCountsbyEntrezGeneIdAndGroup(
+            molecularProfileCaseSets, mutationEventTypes, cnaEventTypes, enrichmentType);
+
+        return alterationEnrichmentUtil.createAlterationEnrichments(alterationCountsbyEntrezGeneIdAndGroup,
+                molecularProfileCaseSets);
+    }
+
+    public Map<String, List<AlterationCountByGene>> getAlterationCountsbyEntrezGeneIdAndGroup(
+        Map<String, List<MolecularProfileCaseIdentifier>> molecularProfileCaseSets,
+        Select<MutationEventType> mutationEventTypes,
+        Select<CopyNumberAlterationEventType> cnaEventTypes,
+        EnrichmentType enrichmentType) {
+        return molecularProfileCaseSets
+            .entrySet()
+            .stream()
```

Comment:
I have copied this logic from the existing code such as f.i. in MutationEnrichmentServiceImpl.jae.

Reply:
yes, if this was the previous business logic [or based on it] it should not be refactored in this PR. We will be doing a survey of performance bottlenecks in the backend soon ... it will be better to test and improve this pattern at that time if we need to.